components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorageMaintenanceWorker.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.sync
+
+import android.content.Context
+import androidx.work.WorkerParameters
+import mozilla.components.concept.storage.StorageMaintenanceWorker
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * A WorkManager Worker that executes [PlacesStorage.runMaintenance].
+ *
+ * If there is a failure or the worker constraints are no longer met during execution,
+ * active write operations on [PlacesStorage] are cancelled.
+ *
+ * See also [StorageMaintenanceWorker].
+ */
+internal class PlacesStorageMaintenanceWorker(context: Context, params: WorkerParameters) :
+    StorageMaintenanceWorker(context, params) {
+
+    val logger = Logger(PLACES_STORAGE_MAINTENANCE_WORKER_TAG)
+
+    override suspend fun operate() {
+        GlobalPlacesDependencyProvider.requirePlacesStorage()
+            .runMaintenance(DB_SIZE_LIMIT_IN_BYTES.toUInt())
+    }
+
+    override fun onError(exception: Exception) {
+        GlobalPlacesDependencyProvider.requirePlacesStorage().cancelWrites()
+        logger.error("An exception occurred while running the maintenance task: ${exception.message}")
+    }
+
+    companion object {
+        private const val IDENTIFIER_PREFIX = "mozilla.components.browser.storage.sync"
+
+        internal const val DB_SIZE_LIMIT_IN_BYTES =
+            75 * 1024 * 1024 // corresponds to 75MiB (in bytes)
+        internal const val WORKER_PERIOD_IN_HOURS = 24L
+        internal const val UNIQUE_NAME = "$IDENTIFIER_PREFIX.PlacesStorageMaintenanceWorker"
+        internal const val PLACES_STORAGE_MAINTENANCE_WORKER_TAG =
+            "$IDENTIFIER_PREFIX.PlacesStorageMaintenanceWorker"
```

Comment:
I like these small bits of thoughtfulness with the addition of the `IDENTIFIER_PREFIX`. 🙂